### PR TITLE
Upgrade scala 212

### PR DESCRIPTION
**Pull request by mrerrormessage**
Changes made during the 6.0-Beta2 release are included as well as changes upgrading to Scala 2.12.

### CODE REVIEW

**Comment by mrerrormessage** [comment]
@bainco , this should be ready for review. None of the models were changed other than to update the version numbers and remove a few newlines at the end of widget strings. The changes made to the sources were made to accommodate the scala 2.12 upgrade:

* Switch http library from play-ws to asynchttpclient. Play-ws wasn't scheduled to be upgraded until early next year and we could consider switching back at that point.
* Upgrade scala-charts
* Switch from creating a separate `diff` process to using DiffUtils (Scala 2.12 has a bug in spawning processes which caused the `diff` process to never return).
* Use Java's ProcessBuilder instead of scala's for calling out to `aspell`. It looks as though the existing method may have been broken for some time since I had to add 2 words to the dictionary to get this test to pass.

File: src/test/resources/modelwords.txt [hunk 1]
```diff
@@ -741,6 +741,7 @@ Cust
 cust
 Custer
 custpop
+customizations
```
Comment (nicolaspayette) [inline src/test/resources/modelwords.txt]: Agreed. Through the years, I've also caught some genuine spelling errors that where mistakenly white-listed here... I suspect there might be a few more of those in there...

File: src/test/scala/org/nlogo/models/SpellCheckTests.scala [hunk 2]
```diff
@@ -27,22 +30,31 @@ class SpellCheckTests extends TestModels with BeforeAndAfterAll {
...
-    "aspell",
-    "--encoding=UTF-8",
-    "--lang=en_US",
-    "--mode=html",
-    "--ignore-case",
-    "--personal", dictPath,
-    "list")
+  def aspell(inPath: Path): (JProcessBuilder, Path) = {
+    val tmpPath = Files.createTempFile("out", ".txt")
+    (new JProcessBuilder("aspell",
```
Comment (nicolaspayette) [inline src/test/scala/org/nlogo/models/SpellCheckTests.scala]: > I always figured that those couple of words that made the tests fail locally (but not on Travis) was some kind of weird dictionary thing.

I thought the same. The dictionary _does_ vary with the Aspell version that's install on the machine running the tests.

File: src/test/scala/org/nlogo/models/LegalInformationTests.scala [hunk 3]
```diff
@@ -23,13 +26,12 @@ class LegalInformationTests extends TestModels {
...
     for {
       notarizedModel <- Try(Notarizer.notarize(model)).toOption
-      inputStream = new ByteArrayInputStream(notarizedModel.getBytes("UTF-8"))
-      diffCommand = Seq("diff", "-u", "-", model.file.getPath)
-      diffs = (diffCommand #< inputStream).lineStream_!.mkString("\n")
-      if diffs.nonEmpty
-    } yield diffs
+      patch = DiffUtils.diff(savedLines, notarizedModel.lines.toSeq.asJava)
+      if ! patch.getDeltas.isEmpty
+    } yield DiffUtils.generateUnifiedDiff(model.file.getPath, model.file.getPath, savedLines, patch, 3).asScala.mkString("\n")
```
Comment (nicolaspayette) [inline src/test/scala/org/nlogo/models/LegalInformationTests.scala]: Agreed!

**Comment by nicolaspayette** [comment]
LGTM too! Thanks for handling this, @mrerrormessage; those are very nice improvements. @bainco, you can merge as far as I'm concerned.

**Pull request merged**